diag_manager/fms_diag_object_container.F90

```diff
@@ -56,13 +56,15 @@ MODULE fms_diag_object_container_mod
    !!
    type, public:: FmsDiagObjectContainer_t
    private
-      TYPE (FmsDlList_t), ALLOCATABLE :: the_linked_list !< This version based on the FDS linked_list.
+      TYPE (FmsDlList_t), pointer :: the_linked_list => null() !< This version based on the FDS linked_list.
```

Comment:
This change makes me worried you are leaking memory

Reply:
No memory leaks in this class. The unit test was checked with Inspector. The class has a procedure decorated with "final" which de-allocates what the class is responsible for.

Reply:
So then why was it changed from `allocatable` to `pointer`?

Reply:
This one wasn't a completely necessary change, but for consistency sake: all links (nodes) between head and tail nodes were pointers anyways, so I made those two also be pointers.